src/main/sensors/gyro.c

```diff
@@ -1053,60 +1065,9 @@ static FAST_CODE FAST_CODE_NOINLINE void gyroUpdateSensor(gyroSensor_t *gyroSens
-            float gyroADCf = gyroSensor->gyroDev.gyroADC[axis] * gyroSensor->gyroDev.scale;
-
-#ifdef USE_GYRO_DATA_ANALYSE
-            gyroADCf = gyroSensor->notchFilterDynApplyFn((filter_t *)&gyroSensor->notchFilterDyn[axis], gyroADCf);
-#endif
-            gyroADCf = gyroSensor->notchFilter1ApplyFn((filter_t *)&gyroSensor->notchFilter1[axis], gyroADCf);
-            gyroADCf = gyroSensor->notchFilter2ApplyFn((filter_t *)&gyroSensor->notchFilter2[axis], gyroADCf);
-            gyroADCf = gyroSensor->lowpassFilterApplyFn((filter_t *)&gyroSensor->lowpassFilter[axis], gyroADCf);
-            gyroADCf = gyroSensor->lowpass2FilterApplyFn((filter_t *)&gyroSensor->lowpass2Filter[axis], gyroADCf);
-            gyroSensor->gyroDev.gyroADCf[axis] = gyroADCf;
-
-            if (!gyroSensor->overflowDetected) {
-                // integrate using trapezium rule to avoid bias
-                accumulatedMeasurements[axis] += 0.5f * (gyroPrevious[axis] + gyroADCf) * sampleDeltaUs;
-                gyroPrevious[axis] = gyroADCf;
-            }
-        }
+        filterGyro(gyroSensor, sampleDeltaUs);
     } else {
-        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            DEBUG_SET(DEBUG_GYRO_RAW, axis, gyroSensor->gyroDev.gyroADCRaw[axis]);
-            // scale gyro output to degrees per second
-            float gyroADCf = gyroSensor->gyroDev.gyroADC[axis] * gyroSensor->gyroDev.scale;
-            // DEBUG_GYRO_SCALED records the unfiltered, scaled gyro output
-            DEBUG_SET(DEBUG_GYRO_SCALED, axis, lrintf(gyroADCf));
-
-#ifdef USE_GYRO_DATA_ANALYSE
-            // apply dynamic notch filter
-            if (isDynamicFilterActive()) {
-                if (axis == X) {
-                    DEBUG_SET(DEBUG_FFT, 0, lrintf(gyroADCf)); // store raw data
-                }
-                gyroADCf = gyroSensor->notchFilterDynApplyFn((filter_t *)&gyroSensor->notchFilterDyn[axis], gyroADCf);
-                if (axis == X) {
-                    DEBUG_SET(DEBUG_FFT, 1, lrintf(gyroADCf)); // store data after dynamic notch
-                }
-            }
-#endif
-            // apply static notch filters and software lowpass filters
-            gyroADCf = gyroSensor->notchFilter1ApplyFn((filter_t *)&gyroSensor->notchFilter1[axis], gyroADCf);
-            gyroADCf = gyroSensor->notchFilter2ApplyFn((filter_t *)&gyroSensor->notchFilter2[axis], gyroADCf);
-            gyroADCf = gyroSensor->lowpassFilterApplyFn((filter_t *)&gyroSensor->lowpassFilter[axis], gyroADCf);
-            gyroADCf = gyroSensor->lowpass2FilterApplyFn((filter_t *)&gyroSensor->lowpass2Filter[axis], gyroADCf);
-            // DEBUG_GYRO_FILTERED records the scaled, filtered, after all software filtering has been applied.
-            DEBUG_SET(DEBUG_GYRO_FILTERED, axis, lrintf(gyroADCf));
-
-            gyroSensor->gyroDev.gyroADCf[axis] = gyroADCf;
```

Comment:
I'd leave this part, and the `for` loop out of the function to keep it minimal.

Reply:
Why not, I'm just a bit reluctant to keep either 2 `for` loops at the top level or having a branch per iteration.

Reply:
I assumed it would be inlined in both cases...

Reply:
It sure would be, but branch is unlikely to go out of the loop.
I see your suggestion as eithe one of those:
```c
if (gyroDebugMode == DEBUG_NONE) {
    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
        filterGyro(gyroSensor, axis, sampleDeltaUs);
    }
} else {
    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
        filterGyroDebug(gyroSensor, axis, sampleDeltaUs);
    }
}
```

```c
for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
    if (gyroDebugMode == DEBUG_NONE) {
        filterGyro(gyroSensor, axis, sampleDeltaUs);
    } else {
        filterGyroDebug(gyroSensor, axis, sampleDeltaUs);
    }
}
```

In the latter case `if` branching is unlikely to be optimized out of the loop, it will likely be performed on each iteration, but I'm not sure without having a look at the resulting assembly.

Reply:
Doh! You are right, obviously.